### PR TITLE
[3.9] Fix the test suite for the old parser

### PR DESCRIPTION
**Pull request by pablogsal**

### CODE REVIEW

**Comment by pablogsal** [comment]
```
./python -Xoldparser ./Tools/scripts/run_tests.py -j 20 -u all -W --slowest --fail-env-changed --timeout=1200
...
== Tests result: SUCCESS ==

412 tests OK.

10 slowest tests:
- test_concurrent_futures: 2 min 32 sec
- test_subprocess: 1 min 56 sec
- test_multiprocessing_spawn: 1 min 19 sec
- test_capi: 1 min 18 sec
- test_asyncio: 1 min 7 sec
- test_gdb: 1 min 6 sec
- test_multiprocessing_forkserver: 1 min 3 sec
- test_unparse: 60.0 sec
- test_tokenize: 55.9 sec
- test_multiprocessing_fork: 54.8 sec

13 tests skipped:
    test_devpoll test_ioctl test_kqueue test_msilib test_ossaudiodev
    test_startfile test_tix test_tk test_ttk_guionly test_winconsoleio
    test_winreg test_winsound test_zipfile64

Total duration: 3 min 11 sec
Tests result: SUCCESS
```

File: Lib/test/test_syntax.py [hunk 1]
```diff
@@ -978,10 +975,11 @@ def func2():
 """
         self._check_error(code, "invalid syntax")
 
+    @unittest.skipIf(support.use_old_parser(), "The old parser ")
```
Comment (isidentical) [inline Lib/test/test_syntax.py]: Why add both `skipIf` and also change the test itself?

**Review by isidentical — APPROVED** [review]
LGTM, thanks for the fix @pablogsal!

File: Lib/test/test_syntax.py (hunk 1 above)
Comment (ambv) [reply Lib/test/test_syntax.py]: This is probably a copypasta on Pablo's part as evidenced by the unfinished message. Let me remove it and see if tests still pass due to the change in the test body.
```suggestion
```

**Comment by ambv** [comment]
Confirmed locally tests still pass with the typo fix. Merging.

**Pull request merged**